operator/bucketcontroller/connector.go

```diff
@@ -48,6 +48,8 @@ func (c *bucketConnector) Connect(ctx context.Context, mg resource.Managed) (man
 		return &NoopClient{}, nil
 	}
 
+	bucket.Status.Endpoint = fmt.Sprintf("objects.%s.cloudscale.ch", bucket.Spec.ForProvider.Region)
+	bucket.Status.EndpointURL = fmt.Sprintf("https://%s", bucket.Status.Endpoint)
```

Comment:
Something else that came to mind:
This location here in the `Connect()` method might not be the ideal location to update the status of a resource.
The [Connect() interface](https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v0.18.0/pkg/reconciler/managed#ExternalConnecter) doesn't directly forbid it, but in [ExternalClient interface](https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v0.18.0/pkg/reconciler/managed#ExternalClient) it's clearly expected that status updates are only expected during `Observe()`. (They explicitly mention that status updates during `Creation()` will be ignored.)
In this case of setting it at `Connect` might just work by accident, but there's a slight risk that Crossplane maintainers might ignore status updates during Connect in the future as well, which can lead to very hard-to-find bugs when we update the runtime. (Note: The opposite could also happen, that it's allowed, but I'm pretty sure they intent that status updates are only done during Observe(), even if updating the endpoints here is not directly related to observing an external resource.

So, it's probably safer that we create a client using something like this:
```go
func getEndpointURL(bucket *cloudscalev1.Bucket) string {
    return fmt.Sprintf("https://objects.%s.cloudscale.ch", bucket.Spec.ForProvider.Region)
}
...
parsed, err := url.Parse(getEndpointURL(bucket))
...
```
and then move these 2 lines to somewhere in `Observe()`

Reply:
Thanks for this input! I've implemented this and it appears to work.